internal/locate/region_request.go

```diff
@@ -2232,7 +2232,7 @@ func (s *RegionRequestSender) onRegionError(
 	// This error is specific to stale read and the target replica is randomly selected. If the request is sent
 	// to the leader, the data must be ready, so we don't backoff here.
 	if regionErr.GetDataIsNotReady() != nil {
-		logutil.Logger(bo.GetCtx()).Warn(
```

Comment:
We used to use this for diagnosis. A better but complex idea might be sampling logs.